sklearn/feature_extraction/text.py

```diff
@@ -1117,21 +1118,26 @@ def fit(self, X, y=None):
         X : sparse matrix, [n_samples, n_features]
             a matrix of term/token counts
         """
+        X = check_array(X, accept_sparse=('csr', 'csc'))
         if not sp.issparse(X):
-            X = sp.csc_matrix(X)
+            X = sp.csr_matrix(X)
+        dtype = X.dtype if X.dtype in FLOAT_DTYPES else np.float64
```

Comment:
For CountVectorizer it still makes sense to support integer dtype

Reply:
Ok but this is the TfidfTransformer and we will output float64 here.

Reply:
So it is!